src/main/java/bio/terra/service/tabulardata/google/BigQueryPdao.java

```diff
@@ -435,6 +436,105 @@ public void createSnapshot(Snapshot snapshot, List<String> rowIds) throws Interr
         snapshotViewCreation(datasetBqDatasetName, snapshotName, snapshot, projectId, bigQuery, bigQueryProject);
     }
 
+    private static final String insertAllLiveViewDataTemplate =
+        "INSERT INTO `<project>.<snapshot>.<dataRepoTable>` " +
+            "(<dataRepoTableId>, <dataRepoRowId>) <liveViewTables>";
+
+    private static final String getLiveViewTableTemplate =
+        // TODO pull insert out and loop thru rest w UNION ()
+            "(SELECT '<tableId>', <dataRepoRowId> FROM `<project>.<dataset>.<liveView>`)";
+
+    private static final String mergeLiveViewTablesTemplate =
+        "<selectStatement1> UNION ALL <selectStatement2>";
+
+    private static final String validateSnapshotSizeTemplate =
+        "SELECT * FROM <project>.<snapshot>.<dataRepoTable>";
+
+    public String createSnaphotTableFromLiveViews(
+        BigQueryProject bigQueryProject,
+        List<DatasetTable> tables,
+        String datasetBqDatasetName) throws InterruptedException {
+
+        String output = "";
+
+        // recursively call the merge template to spool together all the different select statements
+        if (tables.size() == 1) {
+            DatasetTable onlyTable = tables.get(0);
+            TableId liveViewId = TableId.of(datasetBqDatasetName, onlyTable.getName());
+            String liveViewTableName = liveViewId.getTable();
+
+            ST sqlTableTemplate = new ST(getLiveViewTableTemplate);
+            sqlTableTemplate.add("tableId", onlyTable.getId());
+            sqlTableTemplate.add("dataRepoRowId", PDAO_ROW_ID_COLUMN);
+            sqlTableTemplate.add("project", bigQueryProject.getProjectId());
+            sqlTableTemplate.add("dataset", datasetBqDatasetName);
+            sqlTableTemplate.add("liveView", liveViewTableName);
+
+            return sqlTableTemplate.render();
+        }
+
+        for (DatasetTable table : tables) {
+            List<DatasetTable> remainingTables = tables.stream().filter(t -> !t.equals(table)).collect(Collectors.toList());
+            String selectStatement1 = createSnaphotTableFromLiveViews(bigQueryProject, Collections.singletonList(table), datasetBqDatasetName);
+            String selectStatement2 = createSnaphotTableFromLiveViews(bigQueryProject, remainingTables, datasetBqDatasetName);
```

Comment:
I think this recursion makes it tougher to understand what's going on here. As an alternative, I think you could make this tweak:
```java
String mergeLiveViewTablesTemplate = "<selectStatements; separator=\" UNION ALL \">"
```
Then you can either `tables.stream().map` to get a list of statements which you add to the template all at once, or `for (table :  tables)` and add each select statement inside the loop. The body of either the map or the loop would be the logic in your `tables.size() == 1` case above.

If it feels like overkill to use the string-templating in that way, I think you could use:
```java
String.join(" UNION ALL ", selectStatements)
```